backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/s3/RefreshS3ClientCron.java

```diff
@@ -0,0 +1,55 @@
+package ai.verta.modeldb.common.artifactStore.storageservice.s3;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.securitytoken.model.Credentials;
+import java.util.TimerTask;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class RefreshS3ClientCron extends TimerTask {
```

Comment:
let's not use TimerTask for anything. It's a very old API, and we should be using scheduled executors if we want things to happen periodically.

Reply:
Ok, will update it..

Reply:
Yes, I have seen your PR, and all those changes are covered in this PR, where some of them are not relevant because we have changed logic using CronJob.

Reply:
ok. this still should _not_ be being done with a TimerTask. Please replace with usage of an ScheduledExecutor.

Reply:
Any `Runnable` can be scheduled...doesn't need to be a TimerTask, which just happens to implement Runnable.

Reply:
https://github.com/VertaAI/modeldb/pull/3593

Reply:
[Replace TimerTask with Runnable](https://github.com/VertaAI/modeldb/pull/3586/commits/6fdfac3839c1424ba337a455aa8ba835d8ba32ae)